### DISCUSSION TITLE
Next 13 is doing a full page refresh instead of client side navigation on route changes

### DISCUSSION THREAD

**Discussion by lucashfreitas** [vercel/next.js#46295] [Help] [answered]
## The issue

After I upgraded to nextjs 13.1.6, whenever I do `router.push`, what before was just a client-side navigation became a full page reload in my application. As my application is huge this is causing problems as I am using client-side auth and whenever the user changes the page it does a full request to the server and load the authentication all the time. (I have done more tests and confirmed that is happening for other nextjs versions as well - `13.0.0` - but only in the versions deployed on vercel).

This is a odd behavior that I've noticed in the last days in two applications.

## Steps to reproduce:

I have created a reproduction which is available here https://github.com/lucashfreitas/next-reload-issue and the deployed vercel can be accessed here https://next-reload-issue.vercel.app/page1. 

1. import router from `next/navigation`.
2. Do a `router.push` to another page in the `app` directory.
3. Note that next will trigger a `document` request or a full refresh.
4. After confirming the issue, if you run `next build` and `next start` and run the production version locally the issue is not present - atm it's only happening in vercel environment.

Here is a discord thread, a few users have also reported this issue: https://discord.com/channels/752553802359505017/1077904797098856458 

## Environment tested

- Next: 13.1.6 & Next 13.0.0
- app folder: true
- node version: v16.19.0

**Comment by modecode22** [comment]
same issue (next js : "13.1.6")

**Comment by lucashfreitas** [comment]
Quick update - when I tried it on `13.2.1-canary.0` I was not able to reproduce the issue - so it seems this was a real bug? Please see more details on these treads on discord:

- https://discord.com/channels/752553802359505017/1077904797098856458
- https://discord.com/channels/752553802359505017/1075391285218779137/1075391285218779137

**Comment by lucashfreitas** [comment, marked as answer]
Can confirm that the issue seems to be resolved on `13.2.1-canary.0`. Please upgrade and test - I have upgraded all my apps (medium sized) and the issue is gone :)

**Reply by kacyee** [reply]
Confirmed, its fixed!

**Comment by jordanebelanger** [comment]
Same issue with 13.1.6 only happens on vercel preview/prod, it started on prod only, then is now happening on preview as well now. 

On localhost, no issue.

EDIT: seems fixed

**Comment by FengeRVictor** [comment]
We have the same issue on next13.1.1

**Comment by FengeRVictor** [comment]
Everything is good on 13.2.1, thank you !

**Comment by AndrewGardhouse** [comment]
I'm encountering this issue on 14.0.1

**Comment by Farazkhanbwn** [comment]
i'm encountering the same issue on next 14.2 .3
can any one helo how i can manage?
when i move from one page to another page mean the detail page of any item 
then on back whole page reload

i can store my data in redux store for navigation and back to the page then they not reload
i think this approach is not good
how i handle this use case?
